src/haz3lschool/SchoolExercise.re

```diff
@@ -24,6 +24,12 @@ module F = (ExerciseEnv: ExerciseEnv) => {
     hints: list(string),
   };
 
+  [@deriving (show({with_path: false}), sexp, yojson)]
+  type syntax_tests = {
+    s_tests: list(Term.UExp.t => bool),
+    hints: list(string),
```

Comment:
since each test needs a hint, it would be better to have `syntax_tests` just be a `list(syntax_test)` where `type syntax_test = (hint, predicate)` and `type hint = string` and `type predicate = Term.UExp.t => bool`